### PR TITLE
Update go 1.18.1

### PR DESCRIPTION
**Pull request by Ryan-Koch**
## Summary

This updates go to version 1.18.1 following along with the steps described [here](https://transcom.github.io/mymove-docs/docs/backend/guides/how-to/upgrade-go-version/).

### Is there anything you would like reviewers to give additional scrutiny?

I use nix for my local environment and things seemed to work okay there. It'd be helpful to have someone confirm things work okay using not nix.

## Running the thing
* [Instructions for asdf](https://transcom.github.io/mymove-docs/docs/backend/guides/how-to/manage-golang-with-asdf/)
* For nix run this:

```
./nix/update.sh
```

### Things to keep in mind when running the thing
* Make sure that you're seeing the right go version when you run `go version`

```
mymove % go version
go version go1.18.1 darwin/arm64
```

* If you're running goland you may need to change your GOROOT that's set in the app:
image

* If you try to run the pre-commit check for go files using `pre-commit run --all-files golangci-lint` and you get an error like the one below you may just need to do a clean on your pre-commit cache like so, ` pre-commit clean ` . 

```
panic: load embedded ruleguard rules: rules/rules.go:13: can't load fmt

goroutine 1 [running]:
github.com/go-critic/go-critic/checkers.init.22()
        ...golangenv-default/pkg/mod/github.com/go-critic/go-critic@v0.6.2/checkers/embedded_rules.go:46 +0x52c
```

### CODE REVIEW

**Review by reggieriser — CHANGES_REQUESTED** [review]
Looks like there's still one `milmove-app` hash that we need to update.  I also added a nit about the `milmove-cypress` hashes.

I use `asdf` and was able to install/verify 1.18.1, build from scratch and run server tests fine, verify linters still pass, and start the app up and spot check it.  I should be good to approve (at least from the `asdf` perspective) after the hash has been fixed.

File: Makefile [hunk 1]
```diff
@@ -1129,7 +1129,7 @@ pretty: gofmt ## Run code through JS and Golang formatters
 
 .PHONY: docker_circleci
 docker_circleci: ## Run CircleCI container locally with project mounted
-	docker pull milmove/circleci-docker:milmove-app-3d4d7c108724ebcbe7d986d63dcd6fb8a4246633
+	docker pull milmove/circleci-docker:milmove-app-f678d1c6a5e2085f733f5408b7affd628f9f3576
 	docker run -it --rm=true -v $(PWD):$(PWD) -w $(PWD) -e CIRCLECI=1 milmove/circleci-docker:milmove-app-3d4d7c108724ebcbe7d986d63dcd6fb8a4246633 bash
```
[the remark this answers is not in the document]
Comment (Ryan-Koch) [reply Makefile]: Good catch, pushed that update 👍

File: .circleci/config.yml [hunk 2]
```diff
@@ -13,7 +13,7 @@ version: '2.1'
 
 # References for variables shared across the file
 references:
-  circleci-docker: &circleci-docker milmove/circleci-docker:milmove-app-64b1fa9ea8c2c7ed8d5773829c142c30190bc2fb
+  circleci-docker: &circleci-docker milmove/circleci-docker:milmove-app-f678d1c6a5e2085f733f5408b7affd628f9f3576
 
   # the cypress image to use
   cypress: &cypress milmove/circleci-docker:milmove-cypress-64b1fa9ea8c2c7ed8d5773829c142c30190bc2fb
```
[the remark this answers is not in the document]
Comment (Ryan-Koch) [reply .circleci/config.yml]: I kind of like the idea of keeping them in sync, so went ahead and made the change here so they match.

**Review by reggieriser — APPROVED** [review]
LGTM, at least from the `asdf` perspective! 🚀  Thanks for fixing the hashes.

**Pull request merged**